Kernel/FileSystem/Inode.cpp

```diff
@@ -88,19 +88,42 @@ void Inode::will_be_destroyed()
         (void)flush_metadata();
 }
 
+ErrorOr<void> Inode::verify_loop_device_not_attached()
+{
+    return m_loop_device_attached.with([](auto& loop_device_attached) -> ErrorOr<void> {
+        if (loop_device_attached)
+            return EPERM;
+        return {};
+    });
+}
+
 ErrorOr<void> Inode::truncate(u64 size)
 {
     MutexLocker locker(m_inode_lock);
+    TRY(verify_loop_device_not_attached());
     return truncate_locked(size);
 }
 
-ErrorOr<size_t> Inode::write_bytes(off_t offset, size_t length, UserOrKernelBuffer const& target_buffer, OpenFileDescription* open_description)
+ErrorOr<size_t> Inode::prepare_and_write_bytes_locked(off_t offset, size_t length, UserOrKernelBuffer const& target_buffer, OpenFileDescription* open_description)
 {
-    MutexLocker locker(m_inode_lock);
+    VERIFY(m_inode_lock.is_locked());
     TRY(prepare_to_write_data());
     return write_bytes_locked(offset, length, target_buffer, open_description);
 }
 
+ErrorOr<size_t> Inode::write_bytes_from_syscall(off_t offset, size_t length, UserOrKernelBuffer const& target_buffer, OpenFileDescription* open_description)
+{
+    MutexLocker locker(m_inode_lock);
+    TRY(verify_loop_device_not_attached());
```

Comment:
Why are we prohibiting writes to the inode if there's a loop device attached? Could be a nice way to test filesystem corruption!